js/base/Exchange.js

```diff
@@ -1576,6 +1586,27 @@ module.exports = class Exchange {
         return this.signHash (this.hashMessage (message), privateKey.slice (-64))
     }
 
+    signTransaction (rawTransaction, privateKey) {
+        const web3 = new Web3 (new Web3.providers.HttpProvider ('https://ropsten.infura.io/'));
```

Comment:
We should reuse the web3 instance on the exchange class, no need to recreate a new one (see other code related to web3 and how it's enabled, for example, theocean.js). We would also like to avoid hardcoding the RPC provider URL as much as possible.
Another issue with this is that we will have to support it in PHP, and porting web3 to PHP turned out to be tricky...

Reply:
Edited this function to use the existing web3 instance and removed the RPC URL by setting it inside the exchange parameters.

This commit reflects the changes: https://github.com/ccxt/ccxt/pull/4822/commits/3f15438dca0083ad280ceaf8dabde79215050364